build_runner/lib/src/generate/build.dart

```diff
@@ -80,12 +81,13 @@ Future<BuildResult> build(List<BuilderApplication> builders,
       writer: writer,
       onLog: onLog);
   var options = await BuildOptions.create(environment,
-      configKey: configKey,
       deleteFilesByDefault: deleteFilesByDefault,
       failOnSevere: failOnSevere,
       packageGraph: packageGraph,
       logLevel: logLevel,
       skipBuildScriptCheck: skipBuildScriptCheck,
+      overrideBuildConfig:
+          await findBuildConfigOverrides(packageGraph, configKey),
```

Comment:
does this mean we need to expose `findBuildConfigOverrides` from `build_runner_core`? Could `BuildOptions.create` call it instead?

Reply:
No. The intent behind this PR is to allow `findBuildConfigOverrides` to remain in `build_runner`.

Reply:
ah, this is the `package_name.build.yaml` handling? I guess Flutter won't need to support that.

Reply:
This is actually, `build.<config>.yaml` handling. The main issue is actually that we are trying to do as little yaml parsing as possible so the main api in core can only deal with `BuildConfig` objects.

The `--config` flag and semantics of it are something build_runner specific imo, and doesn't belong in core.